src/module/core-modules/napa/timer-wrap.cpp

```diff
@@ -0,0 +1,191 @@
+                    sharedContext->Reset();
+                }
+            }
+        }
+    );
+}
+
+void TimerWrap::SetImmediateCallback(const FunctionCallbackInfo<Value>& args) {
+    auto isolate = Isolate::GetCurrent();
+    HandleScope scope(isolate);
+
+    CHECK_ARG(isolate, args.Length() == 1, "1 argument is required for calling 'SetImmediateCallback'.");
+    CHECK_ARG(isolate, args[0]->IsObject(), "Argument \"timeout\" shall be 'Timeout' type.");
+
+    auto zone = reinterpret_cast<NapaZone*>(WorkerContext::Get(WorkerContextItem::ZONE));
+    if (zone == nullptr) {
+        throw new std::runtime_error("Null zone encountered!");
+    }
+    auto scheduler = zone->GetScheduler().get();
+    if (scheduler == nullptr) {
+        throw new std::runtime_error("Null scheduler encountered!");
+    }
+
+    Local<Object> timeout = Local<Object>::Cast(args[0]);
+    auto sharedTimeout = std::make_shared<Persistent<Object>>(isolate, timeout);
+
+    auto context = isolate->GetCurrentContext();
+    auto sharedContext = std::make_shared<Persistent<Context>>(isolate, context);
+
+    auto immediateCallbackTask = buildTimeoutTask(sharedTimeout, sharedContext);
+
+    auto workerId = static_cast<WorkerId>(
+        reinterpret_cast<uintptr_t>(WorkerContext::Get(WorkerContextItem::WORKER_ID)));
+    scheduler->ScheduleOnWorker(workerId, immediateCallbackTask, SchedulePhase::kImmediatePhase);
+}
+
+
+void TimerWrap::SetTimersCallback(const FunctionCallbackInfo<Value>& args) {
+    auto isolate = Isolate::GetCurrent();
+    HandleScope scope(isolate);
+
+    CHECK_ARG(isolate, args.Length() == 1, "1 argument is required for calling 'SetTimersCallback'.");
+    CHECK_ARG(isolate, args[0]->IsObject(), "Argument \"timeout\" shall be 'Timeout' type.");
+
+    auto zone = reinterpret_cast<NapaZone*>(WorkerContext::Get(WorkerContextItem::ZONE));
+    if (zone == nullptr) {
+        throw new std::runtime_error("Null zone encountered!");
+    }
+    auto scheduler = zone->GetScheduler().get();
+    if (scheduler == nullptr) {
+        throw new std::runtime_error("Null scheduler encountered!");
+    }
+    auto workerId = static_cast<WorkerId>(
+        reinterpret_cast<uintptr_t>(WorkerContext::Get(WorkerContextItem::WORKER_ID)));
+
+    Local<Object> timeout = Local<Object>::Cast(args[0]);
+    auto sharedTimeout = std::make_shared<Persistent<Object>>(isolate, timeout);
+
+    auto context = isolate->GetCurrentContext();
+    auto sharedContext = std::make_shared<Persistent<Context>>(isolate, context);
+
+    Local<Number> after = Local<Number>::Cast(timeout->Get(String::NewFromUtf8(isolate, "_after")));
+    std::chrono::milliseconds msAfter{static_cast<int>(after->Value())};
+    auto sharedTimer = std::make_shared<napa::zone::Timer>(
+        [sharedTimeout, sharedContext, scheduler, workerId]() {
+            auto timerCallbackTask = buildTimeoutTask(sharedTimeout, sharedContext);
```

Comment:
Should we create task object here? Seems it will be executed in the timer's scheduler thread.

Reply:
?? Task currently is the only existing way to be scheduled.

Reply:
To be clear, it just create task, but not run it, when timer call its call back.